src/lib/server/blog/posts.ts

```diff
@@ -61,56 +59,90 @@ export async function fetchBlogPosts(options?: BlogFetchOptions): Promise<BlogOu
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-function cacheBlogPosts(output: any): BlogPost[] {
+async function cacheBlogPosts(output: any): Promise<BlogPost[]> {
   const discussions = output.data.repository.discussions.nodes;
 
+  let users: { [name: string]: string } = {};
+  for (const d of discussions) {
+    const { author } = d;
+    if (author in users) {
+      continue;
+    }
+
+    const fullname = await tryGetGhFullName(author.login);
+    if (fullname !== author.login) {
+      users[author.login] = fullname;
+    }
+  }
   /**
    * No types exist for the discussion object, as the structure is generated
    * based on the GraphQL {@link gql}.
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const posts = discussions.map((discussion: any): BlogPost => {
-    const {
-      title,
-      author,
-      createdAt,
-      lastEditedAt: lastEdited,
-      number: id,
-      bodyHTML: html,
-      url: discussionUrl,
-    } = discussion;
-
-    const url = `/blog/${id}`;
-    const officer = getOfficerByGhUsername(author.login);
-    const authorUrl: string = author.url;
-    const displayname: string = officer?.fullName ?? author.login;
-    const picture: string = officer?.picture ?? author.avatarUrl;
-
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const labels = discussion.labels.nodes.map(({ name }: any) => name);
-
-    const post = {
-      id,
-      url,
-      discussionUrl,
-      title,
-      html,
-      createdAt,
-      lastEdited,
-      labels,
-      author: { url: authorUrl, displayname, picture },
-    };
-
-    return post;
-  });
+  const posts = await Promise.all(discussions.map(async (d: any) => mapDiscussion(d, users)));
 
   cache.setAllPosts(posts);
 
   return posts;
 }
 
-function getOfficerByGhUsername(ghUsername: string): Officer | null {
-  // get author by GitHub username
-  const officer = OFFICERS.find((o) => o.displayName !== undefined && o.displayName === ghUsername);
-  return officer ?? null;
+async function mapDiscussion(
+  discussion: any,
+  users: { [name: string]: string }
+): Promise<BlogPost> {
+  const {
+    title,
+    author,
+    createdAt,
+    lastEditedAt: lastEdited,
+    number: id,
+    bodyHTML: html,
+    url: discussionUrl,
+  } = discussion;
+
+  const url = `/blog/${id}`;
+  const authorUrl: string = author.url;
+  const displayname: string = users[author.login];
+  const picture: string = author.avatarUrl;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const labels = discussion.labels.nodes.map(({ name }: any) => name);
+
+  const post = {
+    id,
+    url,
+    discussionUrl,
+    title,
+    html,
+    createdAt,
+    lastEdited,
+    labels,
+    author: { url: authorUrl, displayname, picture },
+  };
+
+  return post;
+}
+
+// function getOfficerByGhUsername(ghUsername: string): Officer | null {
```

Comment:
seem to work now for some reason, but not exactly, maybe look into fixing this or just commit to removing it entirely?

![image](https://user-images.githubusercontent.com/8981287/195970464-778246c2-fb24-4b93-9115-66dde7da42c9.png)